File: Selenium/Activities/Activity17.java
package examples;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity17 {

	public static void main(String[] args) {

			WebDriver driver = new FirefoxDriver();
			WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(10));
			
			driver.get("https://training-support.net/webelements/selects");
			System.out.println("The page title is : "+ driver.getTitle());
			
			WebElement multiSelect = driver.findElement(By.cssSelector("select.h-80"));
			Select select = new Select(multiSelect);
			
			select.deselectByIndex(0);
			select.selectByVisibleText("HTML");
			for(int i = 3; i<6;i++) {
				select.selectByIndex(i);
			}
			
			List<WebElement> selectedOptions = select.getAllSelectedOptions();
			for(WebElement e : selectedOptions) {
				System.out.println("Selected Options : " + e.getText());
			}
			
			select.selectByValue("nodejs");
			
			select.deselectByIndex(4);
			List<WebElement> selectedOptions1 = select.getAllSelectedOptions();
			for(WebElement e : selectedOptions1) {
				System.out.println("Selected Options : " + e.getText());
			}
			
			driver.quit();

	}

}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select


with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/selects")
    print(f"The page title is : {driver.title}")

    multiSelect = driver.find_element(By.CSS_SELECTOR,"select.h-80")
    select = Select(multiSelect)

    select.deselect_by_index(0);
    select.select_by_visible_text("HTML");
    for i in range(3,6):
        select.select_by_index(i)
    selectedOptions = select.all_selected_options
    for e in selectedOptions:
        print(f"Selected Options : {e.text}")
    select.select_by_value("nodejs")
    select.deselect_by_index(4)
    selectedOptions1 = select.all_selected_options

    for e in selectedOptions1 :
        print(f"Selected Options : {e.text}")
    driver.quit()